modules/binary-tuple/src/main/java/org/apache/ignite/internal/binarytuple/BinaryTupleParser.java

```diff
@@ -504,6 +504,22 @@ public final Period periodValue(int begin, int end) {
         }
     }
 
+    /**
+     * Copies raw value of the specified element to the builder.
+     *
+     * @param builder Builder to copy value to.
+     * @param index Index of the element.
+     */
+    protected void copyRawValue(BinaryTupleBuilder builder, int index) {
+        fetch(index, (idx, begin, end) -> {
```

Comment:
I think we can avoid creation of lambda on every read by moving this method to `BinaryTupleReader`. As downside we will need to make buffer package-private. WDYT?